concordances/handlers.go

```diff
@@ -0,0 +1,257 @@
+	"regexp"
+	"strconv"
+)
+
+const (
+	CONCORDANCES                 = "concordances"
+	ContentTypeJson              = "application/json"
+	UUID_Param                   = "uuid"
+	ErrorMsgJson                 = "{\"message\":\"%s\"}"
+	LogMsg503                    = "Error %s concordances"
+	LogMsg404                    = "Concordances not found"
+	ErrorMsg_BadPath             = "Invalid path."
+	ErrorMsg_BadBody             = "Invalid payload."
+	ErrorMsg_MismatchedConceptId = "Concept uuid in payload is different from uuid path parameter"
+	ErrorMsg_MissingConcoredeIds = "Payload has no concorded uuids to store."
+	ErrorMsg_BadJson             = "Corrupted JSON"
+)
+
+var uuidRegex = regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+
+type ConcordancesRwHandler struct {
+	srv  Service
+	conf AppConfig
+}
+
+func NewConcordanceRwHandler(router *mux.Router, conf AppConfig, srv Service) ConcordancesRwHandler {
+	h := ConcordancesRwHandler{srv: srv, conf: conf}
+	healthchkConfig := &healthConfig{appSystemCode: h.conf.AppSystemCode, appName: h.conf.AppName, port: h.conf.Port, srv: srv}
+	h.registerAdminHandlers(router, healthchkConfig)
+	h.registerApiHandlers(router)
+	return h
+}
+
+func (h *ConcordancesRwHandler) registerApiHandlers(router *mux.Router) {
+	log.Info("Registering API handlers")
+
+	rwHandler := handlers.MethodHandler{
+		"GET":    http.HandlerFunc(h.HandleGet),
+		"PUT":    http.HandlerFunc(h.HandlePut),
+		"DELETE": http.HandlerFunc(h.HandleDelete),
+	}
+	countHandler := handlers.MethodHandler{
+		"GET": http.HandlerFunc(h.HandleCount),
+	}
+
+	router.Handle("/concordances/__count", countHandler)
+	router.Handle("/{concordances}/{uuid}", rwHandler)
+
+	//Invalid path patterns
+	router.Handle("/{concordances}/", http.HandlerFunc(h.HandleBadRequest))
+	router.Handle("/{concordances}", http.HandlerFunc(h.HandleBadRequest))
+	router.Handle("/", http.HandlerFunc(h.HandleBadRequest))
+}
+
+func (h *ConcordancesRwHandler) registerAdminHandlers(router *mux.Router, config *healthConfig) {
+	log.Info("Registering admin handlers")
+	healthService := newHealthService(config)
+	hc := health.HealthCheck{SystemCode: h.conf.AppSystemCode, Name: h.conf.AppName, Description: "Stores concordances in cache and notifies downstream services", Checks: healthService.checks}
+	router.HandleFunc(healthPath, health.Handler(hc))
+	router.HandleFunc(status.PingPath, status.PingHandler)
+	router.HandleFunc(status.GTGPath, status.NewGoodToGoHandler(healthService.gtgCheck))
+	router.HandleFunc(status.BuildInfoPath, status.BuildInfoHandler)
+
+	monitoringRouter := httphandlers.TransactionAwareRequestLoggingHandler(log.StandardLogger(), router)
+	monitoringRouter = httphandlers.HTTPMetricsHandler(metrics.DefaultRegistry, monitoringRouter)
+	http.Handle("/", monitoringRouter)
+}
+
+func (h *ConcordancesRwHandler) HandleGet(rw http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	//400
```

Comment:
Please provide example. 
If I understood you correctly and you mean something like this: 
router.Handle(fmt.Sprintf("%s%s", "/concordances/{uuid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}}"), rwHandler)
This will result in 404 Not Found for all HTTP methods, but the api doc suggests it to be 400